### PR TITLE
Created loading screen example

### LINKED ISSUE #12654: Add a loading screen example
## What problem does this solve or what need does it fill?

Loading and unloading assets on a per-level basis is a simple and generally effective strategy. Commonly, this process is masked from users, using loading screens.

We should teach users how to do this in Bevy.

## What solution would you like?

Requirements:

- use a simple state to track asset loading state: `Loading` vs `FullyLoaded`
- spawn some assets in a scene
- show a loading screen to users before assets are loaded
- swap to that scene once (but not before) all assets are loaded

Stretch goals:

- animate the loading screen
- show progress on the loading screen
- transition between multiple levels (may require #11426 to do nicely), unloading assets from the old level as you do so
- wait for shader pipelines to compile too

## What alternative(s) have you considered?

A more dynamic asset loading solution can work better for e.g. open world games, but is more complex to implement and optimize.

We could also implement a first-party solution for this, but I suspect that the complexity will not be high enough to warrant it and the added flexibility / transparency of doing this as an example will be better.

## Additional context

[bevy_asset_loader](https://github.com/NiklasEi/bevy_asset_loader) is a more sophisticated tool for this. You can see a simple implementation of this pattern [in my archived game Emergence](https://github.com/Leafwing-Studios/Emergence/blob/1dc8ab5050739f8047f24c7a3f60ab091cd46888/emergence_lib/src/asset_management/mod.rs#L53). [iyes_progress](https://github.com/IyesGames/iyes_progress) demonstrates how to track progress and transition states when work is complete.

### PR DESCRIPTION
**Pull request by Remi-Godin**
Allows the user to select a scene to load, then a loading screen is shown until all assets are loaded, and pipelines compiled.

# Objective
- Fixes #12654 

## Solution
- Add desired assets to be monitored to a list.
-  While there are assets that are not fully loaded, show a loading screen.
- Once all assets are loaded, and pipelines compiled, show the scene that was loaded.

### CODE REVIEW

**Comment by Remi-Godin** [comment]
If anyone is able to help, I'm having trouble figuring out a solution to figuring out if all pipelines are compiled. Currently, I count the pipelines that are ready, but I need to know the target pipelines count beforehand. Is there any way to get this number dynamically. Or for a simpler approach, know if all pipelines are loaded regardless of how many there are?

**Comment by Remi-Godin** [comment]
Could not find a good way to get a total sub-asset count for the sake of having a progress indicator, so just kept it down to a simple "Loading..." screen. Otherwise, the method I used should allow for having a progress indicator if loading multiple individual assets. Other than that, it should work properly as a way to wait for all assets to be loaded and pipelines to be compiled.

File: examples/games/loading_screen.rs [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+        Camera3dBundle {
+            transform: Transform::from_xyz(5.0, 5.0, -5.0)
+                .looking_at(Vec3::new(0.0, 0.0, 0.0), Vec3::Y),
+            ..default()
+        },
+        LevelComponents,
+    ));
+
+    // Save the asset into the `loading_assets` vector.
+    let house_scene = asset_server.load("models/HouseInTheWoods/HouseInTheWoods.glb#Scene0");
```
Comment (IceSentry) [inline examples/games/loading_screen.rs:153]: I'd prefer if you used an existing gltf like the fox instead of adding another model to the repo.

Comment (Remi-Godin) [reply examples/games/loading_screen.rs]: Will change for the fox, thanks for the feedback.

**Review by alice-i-cecile — APPROVED** [review]
This is quite useful, and I'm generally happy with the approach taken :) In real games, I suspect that an asset-driven level loading pattern is probably more useful, but I'm not sure how we would nicely demonstrate that here.

**Review by afonsolage — APPROVED** [review]

**Pull request merged**